src/UpdateTracker.js

```diff
@@ -91,6 +101,31 @@ function onmessage({ data }) {
     subscriber(update);
 }
 
+function oncloseFor(host) {
+  return function () {
+    let ws = webSockets[host];
+    delete webSockets[host];
+    reconnectAfterBackoff(ws);
+  };
+}
+
+/** Reconnect WebSocket after a backoff delay */
+async function reconnectAfterBackoff(ws) {
+  if (ws.retry++ < 6) {
+    await ws.backoff;
+    const nextDelay = (ws.delay || 500) * 2;
+    ws.resources.forEach(url => trackResource(url, ws.retry, nextDelay));
+  }
+}
+
+function setUpBackOff(webSocket, retryAttempt, backOffDelay) {
+  Object.assign(webSocket, {
+    delay: backOffDelay,
+    backoff: new Promise(resolve => (setTimeout(resolve, backOffDelay))),
```

Comment:
Given that `backoff` always seems to wait `delay` seconds, is it necessary to assign this as a function to the socket? Could we just do the waiting directly here https://github.com/solid/react-components/pull/30/files#diff-b2ffa2da88905c291381e2f58b66eb35R115 ?

Reply:
Yes. This is a lot clearer moving it out of here.